app/brave_main_delegate.cc

```diff
@@ -197,6 +197,9 @@ bool BraveMainDelegate::BasicStartupComplete(int* exit_code) {
   std::string kVariationsServerURL = BRAVE_VARIATIONS_SERVER_URL;
   command_line.AppendSwitchASCII(variations::switches::kVariationsServerURL,
       kVariationsServerURL.c_str());
+#if defined(OFFICIAL_BUILD)
```

Comment:
this is not a good way to handle this. We want to know it's empty when we build, not when we run it. See https://github.com/brave/brave-core/commit/91aeecd19d5dc3f651ce6d0a296583fd0552e80d#diff-1913917741ddb007a0dcb32770879e181747a891d2398b332b7113d936c26f2bR11

Reply:
@bsclifton @bridiver should we remove the guard in favour of https://github.com/brave/brave-core/pull/7468 or keep as additional stop-gap?

Reply:
I'm good with removing in favor of https://github.com/brave/brave-core/pull/7468 - much better to fail at compile time for sure 😄

Reply:
+1 Makes it difficult to build Brave for release since they builds are successful but the program crashes without `BRAVE_VARIATIONS_SERVER_URL` patched in or this check snipped.